docs/en/operations/settings/settings.md

```diff
@@ -2592,4 +2592,58 @@ Possible values:
 
 Default value: `16`.
 
+## optimize_on_insert {#optimize-on-insert}
+
+Enables or disables data transformation before the insertion, as if merge was done on this block (e.g. replacing, collapsing, aggregating...).
```

Comment:
I think it would be better write just `(according to table engine)` instead of `(e.g. replacing, collapsing, aggregating...)`

Reply:
OK.